contracts/modules/LoanMaintenance.sol

```diff
@@ -317,8 +317,8 @@ contract LoanMaintenance is LoanOpeningsEvents, LoanMaintenanceEvents, VaultCont
 	) external view returns (LoanReturnData[] memory loansData) {
 		EnumerableBytes32Set.Bytes32Set storage set = isLender ? lenderLoanSets[user] : borrowerLoanSets[user];
 
-		uint256 end = count.min256(set.values.length);
-		if (end == 0 || start >= end) {
+		uint256 end = start.add(count).min256(set.length());
+		if (start >= end) {
```

Comment:
there are several instances of `for (uint256 i = end - start; i > 0; i--) {` loop. Consider writing it as 
`for (uint256 i = start; i < end; i++) {` if possible (i.e. it's ok to traverse items from start to end), as it is less error-prone